### DISCUSSION TITLE
Prevent Deleting on Folder Context Menu

### DISCUSSION THREAD

**Discussion by cancan101** [pimcore/pimcore#17049] [Q&A / Support]
Is there anyway to disable / hide the "Delete" option on the context menu that shows when right clicking on an Object in a folder:
<img width="131" alt="image" src="https://github.com/pimcore/pimcore/assets/51059/d2155467-595a-45e6-96a0-f723dc931438">
and this is the code:
https://github.com/pimcore/admin-ui-classic-bundle/blob/1.4/public/js/pimcore/object/folder/search.js#L460-L473

in other places we have a `this.isAllowed("delete")` check (e.g. https://github.com/pimcore/admin-ui-classic-bundle/blob/558c7c245208793ed67a413eca86810d75fcb325/public/js/pimcore/asset/folder.js#L203-L223)

**Comment by markus-moser** [comment]
You can hide the delete button via the workspace permissions of the user. When a user is not allowed to delete the item it will disappear.

If you would need to do this dynamically based on some logic you could use the `pimcore.element.permissions.isAllowed` event (see https://github.com/pimcore/pimcore/blob/11.x/lib/Event/ElementEvents.php). With this it's possible to overrule the base permissions defined in the user configuration.

**Reply by cancan101** [reply]
What in the code here: https://github.com/pimcore/admin-ui-classic-bundle/blob/1.4/public/js/pimcore/object/folder/search.js#L460-L473 would actually hide the Delete action?

The permissions here:
![image](https://github.com/pimcore/pimcore/assets/51059/929bd9be-c383-487a-b061-b8994778c617)
and at the workflow level do not seem to affect the "Delete" shown on the context menu.

**Reply by kingjia90** [reply]
@cancan101 you're right, something was missing, proposing a PR https://github.com/pimcore/admin-ui-classic-bundle/pull/530 to fix that

**Reply by markus-moser** [reply]
Yes, that's a bug!

**Reply by cancan101** [reply]
@mattamon, you mentioned that https://github.com/pimcore/admin-ui-classic-bundle/pull/530 is now obsolete. Was this fixed already? I also had a follow up comment that a related check for "editing" was needed: https://github.com/pimcore/admin-ui-classic-bundle/pull/530#issuecomment-2138297858.

**Reply by mattamon** [reply]
@cancan101 We had an internal discussion. It started as a small PR but more and more problems came up, so the current decision was to not invest more time into this. If you need more details maybe @kingjia90 can give you some insights. Sorry for the confusion

**Reply by kingjia90** [reply]
I am replacing that PR with https://github.com/pimcore/admin-ui-classic-bundle/pull/671 which directly fix the problem initially mentioned here

So far, it is just a visual problem and require refactoring that may not fit in a bug fix (minor release, potentially major), so we agreed to leaving the batch related stuff as is for the moment, as the whole grid would be polished and replaced by new Studio UI

**Reply by kingjia90** [reply]
Meanwhile, if you are able to find a better alternative and consistent solution, and takeover by providing a PR for it, i would be glad to review it.

**Reply by cancan101** [reply]
Do you want me to open a ticket for?:
> You might also need a similar fix to prevent bulk editing in the grid view. I have editing disabled for a given workflow state but am still able to edit the Object through the grid view.

and then:
>if you are able to find a better alternative and consistent solution...

Let me see what I can come up with.

**Reply by kingjia90** [reply]
For the ticket, as you like, if you have some inputs or discussion to do first on how to proceed, otherwise directly in a PR would also work 👍🏻 
Thank you in advance!

**Reply by cancan101** [reply]
https://github.com/pimcore/admin-ui-classic-bundle/pull/671 does not solve my initial problem. It is performing a permission check against user and not the object based on the workflow state. `data.data.permissions.delete` looks to check the user's permissions but isn't as comprehensive as https://github.com/pimcore/admin-ui-classic-bundle/blob/19e6014df22edaaf41999ef2ef6fc892e72ffff4/public/js/pimcore/object/object.js#L1066-L1068 with `'delete'`.  i.e. based on the workflow state, the delete button on the Object will be hidden:
![image](https://github.com/user-attachments/assets/841fa54d-42a7-4ea8-a264-ea27b53c1cee)
but the permission will show `1` for delete so the new logic doesn't hide delete in the context menu.

See: https://github.com/pimcore/admin-ui-classic-bundle/issues/673
